src/request/components/request-detail-panel-execution.jsx

```diff
@@ -336,6 +338,60 @@ var ParamaterList = React.createClass({
     }
 });
 
+var MiddlewareList = React.createClass({
+    render: function() {
+        var middlewareStartMessages = this.props.middlewareStartMessages;
+        var middlewareEndMessages = this.props.middlewareEndMessages;
+        var beforeExecuteCommandMessages = this.props.beforeExecuteCommandMessages;
+        var afterExecuteCommandMessages = this.props.afterExecuteCommandMessages;
+        var mongoDBMessages = this.props.mongoDBMessages;
+        
+        var content = null;
+        
+        if (middlewareStartMessages
+            && middlewareEndMessages
+            && middlewareStartMessages.length >= 0
+            && middlewareEndMessages.length >= 0) {
+        
+            middlewareStartMessages = middlewareStartMessages.sort(function (a,b) { return a.ordinal - b.ordinal; });
+            middlewareEndMessages = _.indexBy(middlewareEndMessages, 'payload.correlationId');
```

Comment:
My concern is that it's not clear that the message ordinal *must* also correspond to the order in which the events occurred, across all of the platforms.  (Consider a case where the message publisher does the publishing off-thread for performance reasons; in that case it may be less likely for the ordinal has any bearing on message ordering within a given context.

I think we may still need some discussion around this to make sure we're doing the right thing.